src/nl/rubensten/texifyidea/settings/TexifySettings.kt

```diff
@@ -34,9 +34,50 @@ class TexifySettings : PersistentStateComponent<TexifySettings> {
     // Index of selected combobox item
     var automaticQuoteReplacement = QuoteReplacement.NONE
 
+    /**
+     * internal list which stores the commands data
+     */
+    private val labelCommandsInternal: HashMap<String, LabelingCommandInformation> =
+            hashMapOf("\\label" to LabelingCommandInformation("\\label", 1, true))
+
+    /**
+     * Verify that any key in the list has a leading backslash
+     */
+    val labelCommands: Map<String, LabelingCommandInformation>
+        get() {
+            return labelCommandsInternal.mapKeys { it.key.addLeadingSlash() }
+        }
+
+    /**
+     * Do not use this field anywhere! It is only needed to persist the data of the commands to the settings.
+     * It maps any command to its string representation and on first load back to the object
+     */
+    var labelCommandsAsString: Map<String, String>
+        get() {
+            return labelCommandsInternal.mapValues { it.value.toSerializableString() }
+        }
+        set(value) {
+            value.forEach { (key, info) -> labelCommandsInternal[key] = LabelingCommandInformation.fromString(info) }
+        }
+
     override fun getState() = this
 
     override fun loadState(state: TexifySettings) {
         XmlSerializerUtil.copyBean(state, this)
     }
+
+    fun addCommand(cmd: LabelingCommandInformation) {
+        labelCommandsInternal[cmd.commandName] = cmd
+    }
+
+    fun removeCommand(cmdName: String) {
+        labelCommandsInternal.remove(cmdName)
+    }
+
+    val labelAnyCommands: Map<String, LabelingCommandInformation>
```

Comment:
I am not sure what this does? How does `AnyCommand` relate to `PreviousCommand`? Add clarification (or change the name)

Reply:
The commands in this list could label previous commands. I changed the name and added an comment for clarification.